scripts/ci-entrypoint.sh

```diff
@@ -79,28 +79,25 @@ select_cluster_template() {
         # shellcheck source=scripts/ci-build-kubernetes.sh
         source "${REPO_ROOT}/scripts/ci-build-kubernetes.sh"
         export CLUSTER_TEMPLATE="test/dev/cluster-template-custom-builds.yaml"
-    elif [[ -n "${CI_VERSION:-}" ]] || [[ -n "${USE_CI_ARTIFACTS:-}" ]] || [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+    elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]] || [[ -n "${CI_VERSION:-}" ]]; then
         # export cluster template which contains the manifests needed for creating the Azure cluster to run the tests
-        GOPATH="$(go env GOPATH)"
-        if ls "${GOPATH}"/src/k8s.io/kubernetes; then
-            KUBERNETES_BRANCH="$(cd "${GOPATH}/src/k8s.io/kubernetes" && git rev-parse --abbrev-ref HEAD)"
-        fi
-        if [[ "${KUBERNETES_BRANCH:-}" =~ "release-" ]]; then
-            CI_VERSION_URL="https://dl.k8s.io/ci/latest-${KUBERNETES_BRANCH/release-}.txt"
-        elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+        export CLUSTER_TEMPLATE="test/ci/cluster-template-prow-ci-version.yaml"
+        if [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
             CI_VERSION_URL="https://dl.k8s.io/ci/${KUBERNETES_VERSION}.txt"
         else
             CI_VERSION_URL="https://dl.k8s.io/ci/latest.txt"
         fi
-        export CLUSTER_TEMPLATE="test/ci/cluster-template-prow-ci-version.yaml"
         export CI_VERSION="${CI_VERSION:-$(curl -sSL ${CI_VERSION_URL})}"
         export KUBERNETES_VERSION="${CI_VERSION}"
     else
         export CLUSTER_TEMPLATE="test/ci/cluster-template-prow.yaml"
+        if [[ "${KUBERNETES_VERSION:-}" =~ "stable" ]]; then
+            CI_VERSION_URL="https://dl.k8s.io/release/${KUBERNETES_VERSION}.txt"
```

Comment:
this needs to be before https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2325/files#diff-2009f7be5363812e22862771f557363795759e67b3ca2e32c50d9fd5f8c5669bR90 since that's where the URL is resolved to get the version... or maybe we just need to add:
```
export KUBERNETES_VERSION="$(curl -sSL ${CI_VERSION_URL})"
```

Reply:
O.K. this is a little more complicated than I anticipated. One observation is that we don't currently support both `KUBERNETES_VERSION` and `CI_VERSION` configuration at the same time. I've added some validation and updated docs, and put the overloaded `CI_VERSION` assignment below all this foo. Are we getting closer?